### PR TITLE
Campaign Kick Endpoint

### LINKED ISSUE #432: Campaign Kick Endpoint
## Description

Campaign owner should be able to kick characters and their corresponding user from their campaign.
The campaign join code should also be  regenerated after a kick request is performed.

## Possible Implementation

Add a destroy function to the campaign registration controller which accepts an array of user ids and removes the user and their characters from the campaign.

### PR DESCRIPTION
**Pull request by RobertoNittolo**
## Description 
Closes #432 
- Add campaign registration destroy route.
- Add campaign registration destroy controller function.
- Add unit tests.

### CODE REVIEW

File: app/Http/Controllers/CampaignRegistrationController.php [hunk 1]
```diff
@@ -53,4 +53,26 @@ public function store(Request $request)
...
+
+        if ($request->has('user_id') && $isCampaignOwner) {
+            $campaignCharacters = $campaign->characters()->whereIn('user_id', $data['user_id'])->get();
+
+            $campaign->users()->detach($data['user_id']);
+            $campaign->characters()->detach($campaignCharacters);
+
+            $campaign->code = $campaign->generateCode();
+            $campaign->save();
+        }
```
Comment (m-triassi) [inline app/Http/Controllers/CampaignRegistrationController.php:74]: pretty good, just missing some error handling here, in cases like this its better to exit early when the state lacks authorization: 
```suggestion
        $user = Auth::user();
        $isCampaignOwner = $campaign->users()->where('user_id', $user->id)->first()->pivot->is_owner;

        if (!$request->has('user_id')) {
            return redirect()->back()->withErrors(['error' => "You need to specify a user."])
        }
        
        if (!$isCampaignOwner) {
            return redirect()->back()->withErrors(['error' => "You don't have permission to do that."])
        }
        
        $campaignCharacters = $campaign->characters()->whereIn('user_id', $data['user_id'])->get();
        
        $campaign->users()->detach($data['user_id']);
        $campaign->characters()->detach($campaignCharacters);

        $campaign->code = $campaign->generateCode();
        $campaign->save();
        
```

File: tests/Feature/Http/Controllers/CampaignRegistrationControllerTest.php [hunk 2]
```diff
@@ -106,4 +106,81 @@ public function store_registers_with_invalid_campaign_code()
...
+        $campaign->characters()->attach($character);
+        $campaign->characters()->attach($character2);
+
+
+        $response = $this->delete(route('campaign-registration.destroy', $campaign), [
+                'user_id' => [$character->user_id, $character2->user_id]
+        ]);
+
+
+
```
Comment (m-triassi) [inline tests/Feature/Http/Controllers/CampaignRegistrationControllerTest.php:141]: just one new line is fine 😂 
```suggestion

```

**Review by m-triassi — CHANGES_REQUESTED** [review]
pretty good, just a couple changes!

File: tests/Feature/Http/Controllers/CampaignRegistrationControllerTest.php (hunk 2 above)
Comment (RobertoNittolo) [reply tests/Feature/Http/Controllers/CampaignRegistrationControllerTest.php]: lol oops😂

File: app/Http/Controllers/CampaignRegistrationController.php (hunk 1 above)
Comment (RobertoNittolo) [reply app/Http/Controllers/CampaignRegistrationController.php]: I agree, I added the changes!

File: routes/web.php [hunk 3]
```diff
@@ -100,6 +100,9 @@
     Route::resource('campaign-registration', App\Http\Controllers\CampaignRegistrationController::class)
         ->only(['create', 'store']);
 
+    Route::delete('/campaign-registration/{campaign}', [App\Http\Controllers\CampaignRegistrationController::class, 'destroy'])
+        ->name("campaign-registration.destroy");
+
```
[the remark this answers is not in the document]
Comment (RobertoNittolo) [reply routes/web.php]: I agree we don't need that, I'll make the changes you suggested!

File: app/Http/Controllers/CampaignRegistrationController.php [hunk 4]
```diff
@@ -53,4 +53,32 @@ public function store(Request $request)
 
         return redirect('campaign.index');
     }
+
+    public function destroy(Request $request, Campaign $campaign_registration)
```
Comment (m-triassi) [inline app/Http/Controllers/CampaignRegistrationController.php:57]: Ah just realized this variable name goes against our code conventions it should be
```suggestion
    public function destroy(Request $request, Campaign $campaignRegistration)
```

Can you rename it everywhere in the method?

**Review by m-triassi — CHANGES_REQUESTED** [review]
one last thing as far as i can tell!

**Review by m-triassi — APPROVED** [review]
Looks great!! nice work :)

**Pull request merged**